src/t_zset.c

```diff
@@ -2795,6 +2800,11 @@ void zunionInterDiffGenericCommand(client *c, robj *dstkey, int numkeysIndex, in
             }
         }
     } else {
+        if (op == SET_OP_INTER && cardinality_only) {
+            addReplyLongLong(c, cardinality);
+            goto cleanup;
+        }
```

Comment:
i'd rather combine this with the `} else {` that's before it, and drop the `cleanup` label.
```
if (dstkey) {
  // store and reply with count
} else if (cardinality_only) {
  // just reply with count
} else {
  // reply with elements
}
```

p.s. we can apply a similar tactics in the other "big" blocks of change in this PR.
i.e. in one you chose to use `goto`, and in others you added an `if` and indentation for a block of a few lines.
in all cases we can use an `else-if` tactic (i'm not sure if it'll be as clean in the other two, but i do feel it's better than the `goto`)